mail_restrict_auto_follower/models/res_partner.py

```diff
@@ -0,0 +1,11 @@
+# -*- coding: utf-8 -*-
+# (c) 2015 Serv. Tecnol. Avanzados - Pedro M. Baeza
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from openerp import models
+
+
+class ResPartner(models.Model):
+    """This is for avoid inheritance problem:
+    https://github.com/odoo/odoo/issues/9084"""
+    _name = 'res.partner'
```

Comment:
@pedrobaeza I started working on a general solution for this: https://github.com/hbrunn/social/blob/8.0-mail_follower_custom_notification/mail_follower_custom_notification/models/base_patch_models_mixin.py

Up to now, I didn't find any weirdnesses, when it's tested more, I'll propose this mixin as an own module for server-tools. But for the time being, you can simply grab a copy and see if it works for you too